packages/payload/src/admin/components/views/Version/RenderFieldsToDiff/fields/Relationship/index.tsx

```diff
@@ -49,9 +49,36 @@ const generateLabelFromValue = (
 
   if (relatedCollection) {
     const useAsTitle = relatedCollection?.admin?.useAsTitle
-    const useAsTitleField = relatedCollection.fields.find(
-      (f) => fieldAffectsData(f) && !fieldIsPresentationalOnly(f) && f.name === useAsTitle,
-    )
+
+    const findFieldRecursively = (fields: Field[], fieldName: string): Field | undefined => {
```

Comment:
I appreciate that you handled title fields that are nested but I'm pretty sure we have this same function somewhere already. If not we should make this a reusable and call it in more places.

Reply:
@DanRibbens Found a reusable function that we already had - `useUseTitleField` - good call - should be all set now